ui/core/components/individual_sim_ui/consumes_picker.ts

```diff
@@ -221,73 +198,123 @@ export class ConsumesPicker extends Component {
+			containerElem: rowElem,
+			options: [
+				{
+					getConfig: () => ConsumablesInputs.makeSpellPowerConsumeInput(
+						relevantStatOptions(ConsumablesInputs.SPELL_POWER_CONFIG, this.simUI),
+						'Arcane',
+					),
+				},
+				{
+					getConfig: () => ConsumablesInputs.makeFirePowerConsumeInput(
+						relevantStatOptions(ConsumablesInputs.FIRE_POWER_CONFIG, this.simUI),
+						'Fire',
+					),
+				},
+				{
+					getConfig: () => ConsumablesInputs.makeFrostPowerConsumeInput(
+						relevantStatOptions(ConsumablesInputs.FROST_POWER_CONFIG, this.simUI),
+						'Frost',
+					),
+				},
+				{
+					getConfig: () => ConsumablesInputs.makeshadowPowerConsumeInput(
+						relevantStatOptions(ConsumablesInputs.SHADOW_POWER_CONFIG, this.simUI),
+						'Shadow',
+					),
+				},
+			],
+			parentElem: spellsCnsumesElem,
+		})
 	}
 
 	private buildEngPicker() {
 		let fragment = document.createElement('fragment');
 		fragment.innerHTML = `
       <div class="consumes-row input-root input-inline">
         <label class="form-label">Engineering</label>
-        <div class="consumes-row-inputs consumes-trade"></div>
+        <div class="consumes-row-inputs consumes-trade">
+				</div>
       </div>
     `;
 
-		this.rootElem.appendChild(fragment.children[0] as HTMLElement);
-
+		const rowElem = this.rootElem.appendChild(fragment.children[0] as HTMLElement);
 		const tradeConsumesElem = this.rootElem.querySelector('.consumes-trade') as HTMLElement;
 
-		buildIconInput(tradeConsumesElem, this.simUI.player, IconInputs.Sapper);
-		buildIconInput(tradeConsumesElem, this.simUI.player, IconInputs.FillerExplosiveInput);
+		this.buildPickers({
+			changeEmitters: [this.simUI.player.levelChangeEmitter, this.simUI.player.professionChangeEmitter],
+			containerElem: rowElem,
+			options: [
+				{
+					getConfig: () => ConsumablesInputs.Sapper,
+				},
+				{
+					getConfig: () => ConsumablesInputs.makeExplosivesInput(
+						relevantStatOptions(ConsumablesInputs.EXPLOSIVES_CONFIG, this.simUI),
+						'Explosives',
+					),
+				}
+			],
+			parentElem: tradeConsumesElem,
+		})
+	}
+
+	private buildPetPicker() {
+		if (!this.simUI.individualConfig.petConsumeInputs?.length) return
 
-		const updateProfession = () => {
-			if (this.simUI.player.hasProfession(Profession.Engineering))
-				tradeConsumesElem.parentElement!.classList.remove('hide');
+		let fragment = document.createElement('fragment');
+		fragment.innerHTML = `
+			<div class="consumes-row input-root input-inline">
+				<label class="form-label">Pet</label>
+				<div class="consumes-row-inputs consumes-pet"></div>
+			</div>
+		`;
+
+		this.rootElem.appendChild(fragment.children[0] as HTMLElement);
+		const petConsumesElem = this.rootElem.querySelector('.consumes-pet') as HTMLElement;
+
+		this.simUI.individualConfig.petConsumeInputs.map(iconInput => buildIconInput(petConsumesElem, this.simUI.player, iconInput));
+	}
+
+	private buildPickers({containerElem, changeEmitters, options, parentElem}: {
```

Comment:
This function allows the pickers to be rebuilt dynamically when the relevant emitters fire while also showing / hiding pickers and the outer containers based on what is shown. This could be cleaned up by moving the additional level/faction/equipment filtering to `relevantStatOptions` but the lift is quite a bit bigger since I'd have to redo pretty much all of the buffs/debuffs helper functions.